lib/resourcemerge/apps.go

```diff
@@ -3,14 +3,16 @@ package resourcemerge
 import (
 	appsv1 "k8s.io/api/apps/v1"
 	"k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/utils/pointer"
 )
 
 // EnsureDeployment ensures that the existing matches the required.
 // modified is set to true when existing had to be updated with required.
 func EnsureDeployment(modified *bool, existing *appsv1.Deployment, required appsv1.Deployment) {
 	EnsureObjectMeta(modified, &existing.ObjectMeta, required.ObjectMeta)
 
-	if required.Spec.Replicas != nil && *required.Spec.Replicas != *existing.Spec.Replicas {
+	ensureReplicasDefault(&required)
+	if existing.Spec.Replicas == nil || (required.Spec.Replicas != nil && *required.Spec.Replicas != *existing.Spec.Replicas) {
```

Comment:
> required.Spec.Replicas != nil

All these can be dropped now with `ensureReplicasDefault`

Reply:
Thanks; I've dropped the `required.Spec.Replicas != nil` guard with f4b194af -> dd827544.